sdk/storage/azure-storage-blob-changefeed/src/test/java/com/azure/storage/blob/changefeed/BlobLazyDownloaderTest.groovy

```diff
@@ -0,0 +1,131 @@
+package com.azure.storage.blob.changefeed
+
+import com.azure.core.util.FluxUtil
+import com.azure.storage.blob.BlobAsyncClient
+import com.azure.storage.blob.models.BlobStorageException
+import com.azure.storage.common.implementation.Constants
+import reactor.core.publisher.Flux
+import spock.lang.Unroll
+
+import java.nio.ByteBuffer
+
+class BlobLazyDownloaderTest extends APISpec {
+
+    BlobAsyncClient bc
+    BlobLazyDownloaderFactory factory
+
+    def setup() {
+        def cc = primaryBlobServiceAsyncClient.getBlobContainerAsyncClient(generateContainerName())
+        cc.create().block()
+        bc = cc.getBlobAsyncClient(generateBlobName())
+        factory = new BlobLazyDownloaderFactory()
+    }
+
+    byte[] downloadHelper(BlobLazyDownloader downloader) {
+        OutputStream os = downloader.download()
+            .reduce(new ByteArrayOutputStream(),  { outputStream, buffer ->
+            outputStream.write(FluxUtil.byteBufferToArray(buffer))
+            return outputStream;
+        }).block()
+        return os.toByteArray()
+    }
+
+    byte[] uploadHelper(int size) {
+        def input = getRandomByteArray(size)
+        def data = Flux.just(ByteBuffer.wrap(input))
+        bc.upload(data, null).block()
+        return input
+    }
+
+    @Unroll
+    def "download blockSize"() {
+        setup:
+        byte[] input = uploadHelper(size)
+
+        when:
+        byte[] output = downloadHelper(new BlobLazyDownloader(bc, blockSize, 0))
+
+        then:
+        output == input
```

Comment:
You might be able to use a `Spy` from Spock to help check that you call download the right number of times.

Reply:
done